spec/src/main/asciidoc/entities.asciidoc

```diff
@@ -18,42 +18,80 @@
 
 === Types vs InputTypes
 
-TODO: info on types and input types
+GraphQL differentiates types from input types.  Input types are entities that are sent by the client as arguments to queries or
+mutations. Types are entities that are sent from the server to the client as return types from queries or mutations.
+
+In many cases the same Java type can be used for input (sent _from_ client) and output (sent _to_ the client), but there are
+cases where an application may need two different Java types to handle input and output.
+
+The `@Type` annotation is used for output entities while the `@InputType` annotation is used for input entities.
+
+Normally these annotations are unnecessary if the type can be serialized and/or deserialized by JSON-B, and if the type is
+specified in a query or mutation method. These annotations can be used to specify the name of the type in the GraphQL schema;
+by default, the entity name in the schema will be the same as the simple class name of the entity type.  These annotations can
+also be used to add a description to the entity in the schema.
 
 === Interfaces as GraphQL entity types
 
-TODO: info on how to use interfaces as types/inputtypes
+It is possible for entities (types and input types) to be defined as a Java interfaces. In order for JSON-B to deserialize an
+interface, the interface may need a `JsonbDeserializer` in order to instantiate a concrete type.
 
 === Limitations
 
 ==== Generic types
 
 TODO: info on limitations to generic types (collections only?)
 
-=== Deprecation
-
-TODO: info on how to use the `@Deprecated` annotation appropriately
-
 === Fields
 
 ==== Scalars
 
 TODO: info on scalars/primitives
 
+=== Deprecation
+
+Application developers can mark entity fields as deprecated in the GraphQL schema with the `@Deprecated` annotation -
+either the annotation that comes from the JDK in the `java.lang` package or the MicroProfile GraphQL annotation in the
+`org.eclipse.microprofile.graphql` package.  The latter annotation allows the developer to specify additional text that
+might provide more information to the consumers of the schema.
+
 ==== Default Values
 
-TODO: info on how to use the `@DefaultValue` annotation appropriately
+The `@DefaultValue` annotation may be used to specify a value in an input type to be used if the client did not specify
+a value. Default values may only be specified on input types (and also as `@Argument` parameters) and will have no
+effect if specified on output types.  The value specified in this annotation may be
 
 ==== FieldsOrder / InputFieldsOrder
 
 TODO: info on fields ordering
 
 ==== Ignorable fields
 
-TODO: info on how to use the `@Ignore` annotation appropriately
+There may be cases where a developer wants to use a class as a GraphQL type or input type, but use fields that should
+not be part of the exposed schema. The `@Ignore` annotation can be placed on the field to prevent it from being part of
+the schema. 
```

Comment:
Should we not also (or only) support @JsonbTransient (https://javaee.github.io/jsonb-spec/users-guide.html#ignoring-properties)

Reply:
Good question.  If we do support `@JsonbTransient` (which works very similar to how `@Ignore` works wrt setting on the field vs the setter vs the getter) and JSON-B is a required dependency of MP GraphQL, then maybe we should remove the `@Ignore` annotation and just use `@JsonbTransient`.  What do you think?

Reply:
Or do what we did with @Deprecated ? Support both ? Maybe merge for now but again, discuss in Gitter or add it to the agenda for next week ?